controller/Lists.php

```diff
@@ -64,38 +64,38 @@
 class Lists extends tao_actions_CommonModule
 {
     use OntologyAwareTrait;
+    use HttpJsonResponseTrait;
 
     private const REMOTE_LIST_PREVIEW_LIMIT = 20;
 
     /** @var bool */
     private $isListsDependencyEnabled;
 
     /**
-     * Show the list of users
+     * This REST endpoint:
+     * - Returns the page with the lists for GET requests
+     * - Creates a new list and returns its name and URI for POST requests
+     *
      * @return void
      */
     public function index()
     {
-        tao_helpers_Scriptloader::addCssFile(Template::css('lists.css', 'tao'));
-        $this->defaultData();
-
-        $myAdderFormContainer = new tao_actions_form_List();
-        $myForm = $myAdderFormContainer->getForm();
-
-        if ($myForm->isSubmited()) {
-            if ($myForm->isValid()) {
-                $values = $myForm->getValues();
-                $newList = $this->getListService()->createList($values['label']);
-                $i = 0;
-                while ($i < $values['size']) {
-                    $this->getListService()->createListElement($newList, __('element') . ' ' . ($i + 1));
-                    $i++;
-                }
+        if ($this->getPsrRequest()->getMethod() === 'POST') {
+            // @todo Move isXmlHttpRequest() check inside ListCreator
+            if (!$this->isXmlHttpRequest()) {
+                throw new common_exception_BadRequest('wrong request mode');
             }
-        } else {
-            $myForm->getElement('label')->setValue(__('List') . ' ' . (count($this->getListService()->getLists()) + 1));
+
+            // @todo Retrieve it through the service locator
+            $creator = new ListCreator($this->getListService());
+            $createdResponse = $creator->createByRequest($this->getPsrRequest());
+
+            $this->setSuccessJsonResponse($createdResponse, 201);
+            return;
```

Comment:
```suggestion
            $this->setSuccessJsonResponse($createdResponse, 201);

            return;
```

Reply:
Fixed in f69bded5ec5315dcad5389bfee9be05111742fe2, thanks!